lmnet/lmnet/datasets/dataset_iterator.py

```diff
@@ -94,7 +95,9 @@ def __init__(self, dataset, result_queue, seed):
         # TODO(tokunaga): the number of processes should be configurable
         self.seed = seed + 1  # seed must not be 0 because using xorshift32.
         self.support_getitem = hasattr(dataset, "__getitem__")
-        self.pool = Pool(processes=8, initializer=_prefetch_setup,
+        self.num_parallel_process = dataset.num_parallel_process \
```

Comment:
`num_parallel_process` is already decided in `DatasetIterator.__init()__`, no need to change in here.
Please change to `def __init__(self, dataset, result_queue, seed, num_parallel_process)` , 
and `self.num_parallel_process = num_parallel_process` in `_MultiProcessDatasetPrefetchThread.__init__()`

Reply:
@ruimashita 
Sorry, I completely misunderstood your point. 
I fixed it.